Mathlib/Order/Concept.lean

```diff
@@ -146,6 +149,25 @@ structure Concept extends Set α × Set β where
 
 initialize_simps_projections Concept (+toProd, -fst, -snd)
 
+/-- The **Dedekind-MacNeille completion** of a partial order is the smallest complete lattice that
+contains it. We define here the type of Dedekind cuts of `α` as the `Concept` lattice of the `≤`
+relation of `α`.
+
+For `A : DedekindCut α`, the sets `A.fst` and `A.snd` are related by `upperBounds A.fst = A.snd` and
+`lowerBounds A.snd = A.fst`.
+
+The file `Order.Dedekind` proves that if `α` is a partial order and `β` is a complete lattice, any
+embedding `α ↪o β` factors through `DedekindCut α`. -/
+abbrev DedekindCut [Preorder α] := Concept α α (· ≤ ·)
+
+@[simp]
+theorem DedekindCut.upperBounds_fst [Preorder α] (A : DedekindCut α) :
+    upperBounds A.fst = A.snd := A.closure_fst
+
+@[simp]
+theorem DedekindCut.lowerBounds_snd [Preorder α] (A : DedekindCut α) :
+    lowerBounds A.snd = A.fst := A.closure_snd
```

Comment:
Could this either come at the very end of the file or go in `Order.DedekindCut`?